R/learner_BART_surv_bart.R

```diff
@@ -0,0 +1,212 @@
+#' @title Survival Bayesian Additive Regression Trees Learner
+#' @author bblodfon
+#' @name mlr_learners_surv.bart
+#'
+#' @description
+#' Fits a Bayesian Additive Regression Trees (BART) learner to right-censored
+#' survival data. Calls [BART::mc.surv.bart()] from \CRANpkg{BART}.
+#'
+#' @details
+#' Two types of prediction are returned for this learner:
+#' 1. `distr`: a 3d survival array with observations as 1st dimension, time
+#' points as 2nd and the posterior draws as 3rd dimension.
+#' 2. `crank`: the expected mortality using [mlr3proba::.surv_return]. The parameter
+#' `which.curve` decides which posterior draw (3rd dimension) will be used for the
+#' calculation of the expected mortality. Note that the median posterior is
+#' by default used for the calculation of survival measures that require a `distr`
+#' prediction, see more info on [PredictionSurv][mlr3proba::PredictionSurv].
+#'
+#' @section Custom mlr3 defaults:
+#' - `mc.cores` is initialized to 1 to avoid threading conflicts with \CRANpkg{future}.
+#'
+#' @section Custom mlr3 parameters:
+#' - `quiet` allows to suppress messages generated by the wrapped C++ code. Is `TRUE` by default.
+#' - `importance` allows to choose the type of importance. Default is `count`, see `importance()` for more details.
+#' - `which.curve` allows to choose which posterior draw will be used for the calculation of the `crank` prediction. By default the **median posterior** is used.
+#'
+#' @templateVar id surv.bart
+#' @template learner
+#'
+#' @references
+#' `r format_bib("sparapani2021nonparametric", "chipman2010bart")`
+#'
+#' @template seealso_learner
+#' @template example
+#' @export
+delayedAssign(
```

Comment:
the delayedAssign is actually not necesary (I am currently removing it for other learners here: https://github.com/mlr-org/mlr3extralearners/pull/301)

Reply:
Ah okay and I wanted to ask you about this! What is it? why there was a need to use it at some point? Let me know if you want it removed here or you will do it in the other PR

Reply:
remove it here please. 

I think this never had the intended effect so explaining it would not be super useful :) (this was a misunderstanding from my side)